projects/packages/classic-theme-helper/src/social-menu/icon-functions.php

```diff
@@ -0,0 +1,213 @@
+}
+
+if ( ! function_exists( 'jetpack_social_menu_social_links_icons' ) ) {
+	/**
+	 * Returns an array of supported social links (URL / regex and icon name).
+	 * For regex, use the # delimiter.
+	 *
+	 * @return array $social_links_icons
+	 */
+	function jetpack_social_menu_social_links_icons() {
+		// Supported social links icons.
+		$social_links_icons = array(
+			'#https?:\/\/(www\.)?amazon\.(com|cn|in|fr|de|it|nl|es|co|ca)\/#' => 'amazon',
+			'500px.com'         => '500px',
+			'apple.com'         => 'apple',
+			'itunes.com'        => 'apple',
+			'bandcamp.com'      => 'bandcamp',
+			'behance.net'       => 'behance',
+			'blogger.com'       => 'blogger',
+			'blogspot.com'      => 'blogger',
+			'bsky.app'          => 'bluesky',
+			'codepen.io'        => 'codepen',
+			'deviantart.com'    => 'deviantart',
+			'discord.gg'        => 'discord',
+			'discordapp.com'    => 'discord',
+			'digg.com'          => 'digg',
+			'dribbble.com'      => 'dribbble',
+			'dropbox.com'       => 'dropbox',
+			'etsy.com'          => 'etsy',
+			'eventbrite.com'    => 'eventbrite',
+			'facebook.com'      => 'facebook',
+			'/feed/'            => 'feed',
+			'flickr.com'        => 'flickr',
+			'foursquare.com'    => 'foursquare',
+			'ghost.org'         => 'ghost',
+			'goodreads.com'     => 'goodreads',
+			'google.com'        => 'google',
+			'github.com'        => 'github',
+			'instagram.com'     => 'instagram',
+			'linkedin.com'      => 'linkedin',
+			'mailto:'           => 'mail',
+			'meetup.com'        => 'meetup',
+			'medium.com'        => 'medium',
+			'nextdoor.com'      => 'nextdoor',
+			'patreon.com'       => 'patreon',
+			'pinterest.'        => 'pinterest',
+			'getpocket.com'     => 'pocket',
+			'ravelry.com'       => 'ravelry',
+			'reddit.com'        => 'reddit',
+			'skype.com'         => 'skype',
+			'skype:'            => 'skype',
+			'slideshare.net'    => 'slideshare',
+			'sms:'              => 'sms',
+			'snapchat.com'      => 'snapchat',
+			'soundcloud.com'    => 'soundcloud',
+			'spotify.com'       => 'spotify',
+			'stackoverflow.com' => 'stackoverflow',
+			'strava.com'        => 'strava',
+			'stumbleupon.com'   => 'stumbleupon',
+			'telegram.me'       => 'telegram',
+			'threads.net'       => 'threads',
+			'tiktok.com'        => 'tiktok',
+			'tumblr.com'        => 'tumblr',
+			'twitch.tv'         => 'twitch',
+			'twitter.com'       => 'twitter',
+			'vimeo.com'         => 'vimeo',
+			'vk.com'            => 'vk',
+			'whatsapp.com'      => 'whatsapp',
+			'woocommerce.com'   => 'woocommerce',
+			'wordpress.org'     => 'wordpress',
+			'wordpress.com'     => 'wordpress',
+			'yelp.com'          => 'yelp',
+			'x.com'             => 'x',
+			'xanga.com'         => 'xanga',
+			'youtube.com'       => 'youtube',
+		);
+
+		/*
+		 * Add Mastodon instances to this array.
+		 */
+		$mastodon_instance_list = jetpack_mastodon_get_instance_list();
```

Comment:
This function seems to be coming from [jetpack plugin global functions](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/functions.global.php#L524). 
And is what phan is marking as PhanUndeclaredFunction.

I don't think this will be accessible if the package is used in a different plugin than jetpack

Reply:
Yes I noticed that too - I could presumably copy the function over if needed, I haven't looked into it much though just yet. I won't be dealing with Phan issues until the next PR (I made some minor fixes here, added that suppression, but can't test properly until I'm requiring the code from the package).